rust/src/storage/s3/mod.rs

```diff
@@ -405,80 +470,103 @@ pub trait LockClient: Send + Sync + Debug {
     /// Attempts to acquire lock. If successful, returns the lock.
     /// Otherwise returns [`Option::None`] which is retryable action.
     /// Visit implementation docs for more details.
-    async fn try_acquire_lock(&self) -> Result<Option<LockItem>, StorageError>;
+    async fn try_acquire_lock(&self, data: &str) -> Result<Option<LockItem>, StorageError>;
 
     /// Returns current lock from DynamoDB (if any).
     async fn get_lock(&self) -> Result<Option<LockItem>, StorageError>;
 
+    /// Update data in the upstream lock of the current user still has it.
+    /// The returned lock will have a new `rnv` so it'll increase the lease duration
```

Comment:
```suggestion
    /// The returned lock will have a new `rvn` so it'll increase the lease duration
```

😆 - hoping my PR review contribution eventually amounts to more than doc edits

Reply:
anyway that's a good ones, nobody likes to read some typos and get confused :)